source/common/tls/context_config_impl.cc

```diff
@@ -25,33 +25,36 @@ namespace Tls {
 
 namespace {
 
-std::vector<Secret::TlsCertificateConfigProviderSharedPtr> getTlsCertificateConfigProviders(
+std::vector<TlsCertificateConfigProviderWithName> getTlsCertificateConfigProviders(
     const envoy::extensions::transport_sockets::tls::v3::CommonTlsContext& config,
     Server::Configuration::TransportSocketFactoryContext& factory_context,
     absl::Status& creation_status) {
-  std::vector<Secret::TlsCertificateConfigProviderSharedPtr> providers;
+  std::vector<TlsCertificateConfigProviderWithName> providers;
   if (!config.tls_certificates().empty()) {
+    int unnamed_cert_idx = 0;
     for (const auto& tls_certificate : config.tls_certificates()) {
       if (!tls_certificate.has_private_key_provider() && !tls_certificate.has_certificate_chain() &&
           !tls_certificate.has_private_key() && !tls_certificate.has_pkcs12()) {
         continue;
       }
-      providers.push_back(
+      providers.push_back(TlsCertificateConfigProviderWithName{
+          absl::StrCat("unnamed_cert_", unnamed_cert_idx++),
```

Comment:
I'm worried that these names may conflict with another tls context that also has unnamed certs. Could we use something like the serial number or a hash of the cert, to prevent collisions?